packages/app/src/cli/services/app-logs/ui/components/Logs.tsx

```diff
@@ -0,0 +1,177 @@
+import {subscribeProcess, SubscribeOptions, PollOptions, LogsProcess} from '../../processes/polling-app-logs.js'
+import React, {FunctionComponent, useRef, useState, useEffect} from 'react'
+
+import {Static, Box, Text} from '@shopify/cli-kit/node/ink'
+
+export interface LogsProps {
+  logsProcess: LogsProcess
+  subscribeOptions: SubscribeOptions
+  pollOptions: PollOptions
+}
+
+interface DetailsFunctionRunLogEvent {
+  input: string
+  inputBytes: number
+  invocationId: string
+  output: string
+  outputBytes: number
+  logs: string
+  functionId: string
+  fuelConsumed: string
+  errorMessage: string | null
+  errorType: string | null
+  status: string
+  source: string
+  eventType?: string
+}
+
+const POLLING_INTERVAL_MS = 450
+const POLLING_ERROR_RETRY_INTERVAL_MS = 5 * 1000
+const POLLING_THROTTLE_RETRY_INTERVAL_MS = 60 * 1000
+const ONE_MILLION = 1000000
+
+const Logs: FunctionComponent<LogsProps> = ({
+  logsProcess,
+  pollOptions: {cursor = '', jwtToken, filters},
+  subscribeOptions: {developerPlatformClient, storeId, apiKey},
+}) => {
+  const pollingInterval = useRef<NodeJS.Timeout>()
+  const currentIntervalRef = useRef<number>(POLLING_INTERVAL_MS)
+  const [logs, setLogs] = useState<DetailsFunctionRunLogEvent[]>([])
+  const [errorsState, setErrorsState] = useState<string[]>([])
+  const [jwtTokenState, setJwtTokenState] = useState<string | null>(jwtToken)
+
+  const pollLogs = async (currentCursor: string) => {
+    try {
+      if (jwtTokenState === null) {
+        const jwtToken = await subscribeProcess({
+          developerPlatformClient,
+          storeId,
+          apiKey,
+        })
+        if (!jwtToken) {
+          return
+        }
+        setJwtTokenState(jwtToken.jwtToken)
+      }
+      const {
+        cursor: newCursor,
+        errors,
+        appLogs,
+      } = await logsProcess({jwtToken: jwtTokenState || jwtToken, cursor: currentCursor, filters})
+      if (errors) {
+        if (errors.some((error) => error.includes('429'))) {
+          currentIntervalRef.current = POLLING_THROTTLE_RETRY_INTERVAL_MS
```

Comment:
I'm assuming some of this -- the polling throttle, formatting the event, should be shared between this and what `dev` outputs on a function run?

Reply:
No for the output, the output for the app logs will be different between `dev` and `logs` based on the current [DX](https://docs.google.com/document/d/1R_qB9-n0ahIiEkObNO7_aioWvqByNlJsKHj0_4UrxMA/edit#heading=h.ol4bdsjg0ajm).  

For the polling throttle, we handle that in `dev` [here](https://github.com/Shopify/cli/blob/main/packages/app/src/cli/services/app-logs/poll-app-logs.ts#L53-L75), and we recently updated this to use the `outputWarns` and `outputDebug` so perhaps we can do something similar instead. But with this PR's approach, we show these messages conditally (rather than always outputted to the logs). 🤔